Automattic/wp-calypso
# Jetpack Plans: Add redirect middleware for partner coupon handling

#### Changes proposed in this Pull Request

This PR aims adds a middleware which redirects customers directly to checkout if there is a Jetpack partner coupon present in the redirectUri (param: `partnerCoupon`).

#### Testing instructions

* Setup this branch
  * Spin up Calypso locally with this branch
  * Use the calupso.live domain (`https://container-naughty-einstein.calypso.live/` instead of `http://calypso.localhost:3000/`)
* "Fake" a normal connection flow by going to `http://calypso.localhost:3000/jetpack/connect/plans/**{DOMAIN}**?redirect=https%3A%2F%2F**{DOMAIN}**%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack%26partnerCoupon%3DJPTST_JPTA_123AB` and replace the two instances of `**{DOMAIN}**` with an actual domain (e.g.: `example-site.jurassic.ninja`).
* If you're logged in: Hopefully you didn't see the plans page but instead got directed straight to checkout.
* If you're NOT logged in: you were asked to log in and afterwards directed straight to checkout, instead of the typical plans page.



#### Related to:
* pbtFPC-2c3-p2
* https://github.com/Automattic/jetpack/pull/21813
